Sources/Path.swift

```diff
@@ -81,6 +81,31 @@ class PathSearch: Visitor {
 }
 
 public extension Node {
+    /**
+     Selects nodes based on a path relative to this node. For example, in the
+     following document:
+
+     ```xml
+     <a id="1">
+       <b id="2">
+         <c id="3"/>
+       </b>
+       <c id="4">
+         <d id="5"/>
+       </c>
+     </a>
+     ```
+
+     evaluating the path `["a", "b", "c"]` would select the `<c>` element with
+     `id="3"` but not the `<c>` element with `id="4"`.
```

Comment:
Starting at the document object (the parent of the root `<a>` element), select all children with `nodeName == "a"`. From that set of nodes (with `nodeName == "a"`), select all children with `nodeName == "b"`. Finally, from that set of nodes (with `nodeName == "b"` that are children of nodes with `nodeName == "a"`), select all children with `nodeName == "c"`.